### DISCUSSION TITLE
HomeKit/Siri Functionality?

### DISCUSSION THREAD

**Discussion by stephenmetzger** [mikebrady/shairport-sync#1275] [Q&A] [answered]
I see in the docs that HomeKit and Siri are listed as having incomplete functionality. Is there any functionality? And if so, how do I configure for HomeKit/Siri. Is there a homebridge plugin needed? My Shairport w/ AP2 instances are working great, would love to be able to see them in HomeKit like my Apple TVs. 

Thanks.

**Comment by charlesomer** [comment, marked as answer]
You should be able to add Shairport Sync to the Home app, but currently the metadata doesn't fully work. It may then allow you to use Siri with Shairport Sync too

**Reply by stephenmetzger** [reply]
Thanks for the reply. When playing music to a shairport receiver, it will show up if I go to "+"--> "Add Accessory" --> "I Don't Have a Code or Cannot Scan". But after selecting the speaker, I get the error "Unable to Add Accessory".

**Reply by charlesomer** [reply]
Hmm perhaps try restarting your iPhone, sometimes that can fix things. If not, which version are you running? (`shairport-sync -V`)

**Reply by stephenmetzger** [reply]
Yeah, tried to restart phone. I'm running:
4.0-dev-179-g2184f1b-AirPlay2-OpenSSL-Avahi-ALSA-soxr-metadata-mqtt-dbus-sysconfdir:/etc

**Reply by charlesomer** [reply]
You probably need to update, current latest version is `4.0-dev-311` 😄

**Reply by stephenmetzger** [reply]
Ahhh. I didn’t see that. I’ll give it a shot. Any quick/easy way to update? Or just have to pull the latest from GitHub and rebuild? Prior to the AP2 build I was running in docker... have forgotten the old ways.

**Reply by charlesomer** [reply]
Try `mikebrady/shairport-sync:unstable-development` if you like docker 😉

**Reply by stephenmetzger** [reply]
Dang! Where is all of this info documented. I thought I was up to date on everything. Had no idea that there was an updated or that there's a docker option for AP2, Is the docker install any more or less stable than running as a service? Thanks a ton. I know what I'm up to this weekend...

**Reply by charlesomer** [reply]
Haha! Hopefully on the website at some point (WIP). I only use the docker image and it's been pretty much perfect, I don't think it's less stable 😄 

There's some notes on the docker build here: https://github.com/mikebrady/shairport-sync/tree/development/docker. It's mainly for if you wanted to manually build an image though.

There's an example compose file here: https://github.com/mikebrady/shairport-sync/blob/development/docker/docker-compose.yaml. Which is probably more useful 😊

**Reply by charlesomer** [reply]
In _theory_, you might be able to just update the old docker tag to the `unstable-development` one - depending on your setup...

**Reply by stephenmetzger** [reply]
Thanks a ton for all your help. All are updated and integrated into HomeKit.

**Reply by charlesomer** [reply]
No problem! Glad you were able to get it working 😄

**Reply by noelhibbard** [reply]
> Yeah, tried to restart phone. I'm running:
> 4.0-dev-179-g2184f1b-AirPlay2-OpenSSL-Avahi-ALSA-soxr-metadata-mqtt-dbus-sysconfdir:/etc

I had this problem when first trying to add shairport-sync to HomeKit. I think I restarted my phone but may have also restarted my HomeKit Hub (Apple TV 4 in my case).

Once added you should be able to use Siri to add remove rooms to playback. For example:
"Hey Siri, playback in the Livingroom"
Or:
"Hey Siri, add the Porch to playback"

Don't quote me on the exact commands. I spent a bunch of time trying different commands and it's really finicky. So I sort of gave up on using Siri for AirPlay2 other than telling it to change the volume in the bathroom while I'm in the shower. Sometimes it will add a room to playback and other times it stops playback in all rooms and then switches to the room I told it to add. I don't think this is a limitation of shairport-sync though. I think it's just the typical Siri bugginess. I have a bunch of HomeKit enabled stuff in my house and Siri is supper inconsistent.

"Hey Siri, goodnight"
"Some devices aren't responding"
All the devices actually responded.

"Hey Siri, water the front yard for 1 hour"
"Okay, I stopped watering the front yard"
The sprinklers in the front yard turn on.

It's a mess.

**Reply by noelhibbard** [reply]
If anyone manages to figure out exactly what the Siri commands are to add or remove a specific room please share. I think it would be great to gather up all the useful Siri commands related to AirPlay2 and add them to one of the readmes.

**Comment by mikebrady** [comment]
Just FYI, there's been some improvement in the Home and Siri integration in the latest development version -- `4.1-dev`. Comments welcome!

**Reply by noelhibbard** [reply]
Just curious, should we remove our existing shairport-sync accessories from HomeKit and then re-add them?

**Reply by mikebrady** [reply]
Hi Noel. I think it's a good idea, yeah.

**Reply by acresp** [reply]
I know that there's lot of guesswork involved in this, but hope you'll be able to find how to handle the "Playing" state. Just tried the last rev and it's the same behaviour. The tile in the home app stay greyed out on every devices even if I'm playing something. I think that for instance, on Sonos devices, the tile becomes "lighted up" with the "Playing" state.
But as this, it's all bonus! So great to have the AP2 quiet stable for a moment on all my Pi's ! 🙂

**Reply by stephenmetzger** [reply]
Thanks Mike! Is there a link anywhere for where the changelogs for the dev builds? I'm having trouble keeping up with new releases.

**Reply by mikebrady** [reply]
There's the rather sketchy [RELEASENOTES-DEVELOPMENT.md](https://github.com/mikebrady/shairport-sync/blob/development/RELEASENOTES-DEVELOPMENT.md).

**Reply by stephenmetzger** [reply]
This is exactly what I was looking for. Thank you.
